map/extrapolation/extrapolator.cpp

```diff
@@ -0,0 +1,162 @@
+#include "map/extrapolation/extrapolator.hpp"
+
+#include "geometry/distance_on_sphere.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+
+#include <chrono>
+#include <memory>
+
+namespace
+{
+uint64_t constexpr kMaxExtrapolationTimeMs = 1000;
+uint64_t constexpr kExtrapolationPeriodMs = 200;
+double constexpr kMaxExtrapolationSpeedMPerS = 120.0;
+
+double LinearExtrapolationOfOneParam(double param1, double param2, uint64_t timeBetweenPointsMs,
```

Comment:
Also, your current usage looks misleading to me: you return (x3-x2) where I expect x3.

Reply:
If I name the method Delta, will it be ok with you?

Reply:
Done

Reply:
I wrote LinearExtrapolator instead.